Source/LinqToDB/Linq/Builder/EagerLoading.cs

```diff
@@ -1433,7 +1433,8 @@ internal class KeyDetailEnvelope<TKey, TDetail>
 
 			correctedExpression = expr.Transform(e =>
 			{
-				if (e.NodeType == ExpressionType.MemberAccess)
+				if (e.NodeType == ExpressionType.MemberAccess || e.NodeType == ExpressionType.Call &&
+					((MethodCallExpression)e).IsSameGenericMethod(ParameterContainer.GetValueMethodInfo))
```

Comment:
On quick read, current is not obvious that `IsSameGenericMethod()` belongs to just the `Call` check. 
```suggestion
				if (e.NodeType == ExpressionType.MemberAccess 
				    || e.NodeType == ExpressionType.Call && ((MethodCallExpression)e).IsSameGenericMethod(ParameterContainer.GetValueMethodInfo))
```